### ISSUE TITLE
AJAX navigation

### ISSUE THREAD

**Issue by Viiprogrammer** [ankitskvmdam/clean-jsdoc-theme#136]
**Is your feature request related to a problem? Please describe.**

No

**Describe the solution you'd like**

Add [AJAX](https://ru.wikipedia.org/wiki/AJAX) navigation for menu / `@link` and other links

**Additional context**

It's a little more convenient than page reload transitions, and on mobile it's faster and a little more economical.

**Comment by ankitskvmdam** [comment]
@Viiprogrammer Could you explain this a little bit more?

**Comment by Viiprogrammer** [comment]
With ajax navigation no page reload needed (when you click to the link it load content of page / page with `fetch` or `XMLRequestHttp` and insert getted HTML to current page HTML, after this push new url to browser history.



Examples of realization:

https://habr.com/ru/post/154617/ (available in Russian only :( ) 

https://stackoverflow.com/questions/2586661/url-navigation-in-an-ajax-based-website

https://developer.mozilla.org/en-US/docs/Web/API/History_API/Example

**Comment by ankitskvmdam** [comment]
Oh.. I think it is something like SPA (Single Page Application).

Do you have any opinion/recommendation on how to implement this?

**Comment by Viiprogrammer** [comment]
yes, this technology was before spa and even now it is, where there is no spa, but unlike spa, it is usually there where ssr is used and it is easier to integrate it into any application, it is not necessary to rewrite it on spa

the only thing that is possible after <<go to new page>> with ajax is to call a couple of functions to initialize code highlighting and other things for the new HTML

**Comment by Viiprogrammer** [comment]
yes, this technology was before spa and even now it is, where there is no spa, but unlike spa, it is usually there where ssr is used and it is easier to integrate it into any application, it is not necessary to rewrite it on spa

the only thing that is possible after <<go to new page>> with ajax is to call a couple of functions to initialize code highlighting and other things for the new HTML

**Comment by ankitskvmdam** [comment]
@Viiprogrammer does it mean that now we have to export files in `.php`, if so then it will change the overall workflow.

I have very limited (or no) knowledge of ajax navigation.

**Comment by Viiprogrammer** [comment]
I thought to do it without a backend (without php, node...) because there is no need for a backend, it will greatly complicate life) and will not give any profit

maybe it has some other name besides ajax/pjax, but I never thought about it

**Comment by ankitskvmdam** [comment]
@backspaces @Viiprogrammer I am working on a new version of clean-jsdoc-theme in which I will use next js. Do you have any opinions?

I have created a repo to test the possibilities. https://github.com/ankitskvmdam/clean-jsdoc-next

**Comment by backspaces** [comment]
I don't use React so don't know it's advantages. 11ty is so dead simple yet quite powerful especially for multi page apps. Otoh if you are looking for ready made components, React may be a win,

**Comment by Viiprogrammer** [comment]
> @backspaces @Viiprogrammer I am working on a new version of clean-jsdoc-theme in which I will use next js. Do you have any opinions?
> 
> I have created a repo to test the possibilities. https://github.com/ankitskvmdam/clean-jsdoc-next

I was thinking about Vue / React options, but they have a problem in that they need SSR if you need to get Server side render and at the same time keep what SPA has.

Maybe It's possible to get both without actually rendering on the server side with [Svelte](https://kit.svelte.dev/docs/adapter-static)
But I have never used Svelt and don't know for sure if there are any problems

**Comment by ankitskvmdam** [comment]
@backspaces I looking to directly port [Nextra](https://nextra.site/docs/docs-theme/start) or something similar to this.

@Viiprogrammer There is an [option](https://nextjs.org/docs/pages/building-your-application/deploying/static-exports) in Next.JS to directly export all the static files.

The biggest advantage of using React is its ecosystem.

**Comment by Viiprogrammer** [comment]
> @backspaces I looking to directly port [Nextra](https://nextra.site/docs/docs-theme/start) or something similar to this.
> 
> @Viiprogrammer There is an [option](https://nextjs.org/docs/pages/building-your-application/deploying/static-exports) in Next.JS to directly export all the static files.
> 
> The biggest advantage of using React is its ecosystem.

In any case, a pretty good idea, I think it would be cool
